File: src/main/java/com/leet/easy/ValidParentheses.cpp
#include <stack>
#include <string>
using namespace std;

class Solution
{
public:
	bool isValid(string s)
	{
		stack<char> chars;
		for (auto ch : s)
		{
			if (ch == '(' || ch == '{' || ch == '[')
			{
				chars.push(ch);
			}
			else
			{
				if (chars.size() == 0)
					return false;
				char top = chars.top();
				if (ch == ')' && top != '(')
					return false;
				if (ch == '}' && top != '{')
					return false;
				if (ch == ']' && top != '[')
					return false;
				chars.pop();
			}
		}

		return chars.size() == 0;
	}
};

File: src/main/java/com/leet/easy/ValidParentheses.py
class Solution:
    def isValid(self, s: str) -> bool:
        stack = []
        for e in s:
            if e in ['(', '{', '[']:
                stack.append(e)
            else:
                if len(stack) == 0:
                    return False
                elif e == ']' and stack[-1] != '[':
                    return False
                elif e == '}' and stack[-1] != '{':
                    return False
                elif e == ')' and stack[-1] != '(':
                    return False
                stack.pop()
        return len(stack) == 0
